ios/RNAccengageModule/RNAccengageModule.m

```diff
@@ -98,6 +103,120 @@ @implementation RNAccengageModule
 	[Accengage trackLead:label value:value];
 }
 
+#pragma mark - Get Inbox Messages
+
+RCT_EXPORT_METHOD(
+    getInboxMessages:(RCTResponseSenderBlock)callback
+                     rejecter:(RCTPromiseRejectBlock)reject
+){
+
+    //Get Accengage Inbox
+    [self getAccengageInboxWithSuccess:^(BMA4SInBox *inbox) {
+        _inboxMessageList = inbox;
+        
+        //get message list
+        [self getAccengageMessagesWithSuccess:^(NSMutableArray* messages) {
+            callback(messages);
+        } failure:^(NSString *code, NSString *message, NSError *error) {
+            reject(code,message,error);
+        }];
+        
+    } failure:^(BMA4SInBoxLoadingResult result) {
+        NSString *operation = (result == BMA4SInBoxLoadingResultCancelled ? @"Cancelled" : @"Failed");
+        NSString *errorMessage = [NSString stringWithFormat:@"Inbox loading result had been %@",operation];
+        NSString *operationCode = [NSString stringWithFormat:@"%@",@(result)];
+        reject(operationCode,errorMessage,nil);
+    }];
+}
+
+
+//Get Accengage inbox
+//@success BMA4SInBox
+//@failure BMA4SInBoxLoadingResult
+//
+- (void)getAccengageInboxWithSuccess:(void (^)(BMA4SInBox *inbox))success failure:(void (^)(BMA4SInBoxLoadingResult result))failure
+{
+    [BMA4SInBox obtainMessagesWithCompletionHandler:^(BMA4SInBoxLoadingResult result, BMA4SInBox *inbox) {
+        if(result != BMA4SInBoxLoadingResultLoaded)
+        {
+            failure(result);
+        }else{
+            success(inbox);
+        }
+    }];
+}
+
+//Get Message list
+//@success return a maximun of 10 messages
+//@failure BMA4SInBoxLoadingResult
+//
+- (void)getAccengageMessagesWithSuccess:(void (^)(NSMutableArray *messages))success failure:(RCTPromiseRejectBlock)failure
+{
+    _messageList = [NSMutableArray new];
+    
+    if(_numLoadedMessages < _inboxMessageList.size)
+    {
+        int unreadMessagesCount = (int)(_inboxMessageList.size - _numLoadedMessages);
+        _numLoadedMessages = MIN(unreadMessagesCount, 10);
+        
+        for (int i = 0; i < _numLoadedMessages; i++)
+        {
+            [_messageList addObject:[NSNull null]];
+            [self getInboxMessageAtIndex:i+1 messageCallback:^(NSArray *response) {
+                //override object at index
+                [_messageList setObject:response.firstObject atIndexedSubscript:i];
+                
+                //if the message is the last of the array
+                if(_messageList.count == i)
```

Comment:
This will always trigger when the first message is loaded, i will be the _numLoadedMessages at that point. And _messageList.count will always be that amount as well.

Reply:
I changed this for this:
              if(batchSize - 1 == i)

and the batchSize is:
        int unshownMessages = (int)(_inbox.size - _numLoadedMessages);
        int batchSize = _numLoadedMessages + MIN(unshownMessages, 10);